### ISSUE TITLE
missing interface fields doc comments

### ISSUE THREAD

**Issue by setanarut** [andygeiss/ecs#12]
Can you explain what their purpose `Run()`, `Setup()`, `Teardown()`, `Tick()`, `Setup()`, `Process()`, `Teardown()` for beginners?

```Go
type Engine interface {
        // ???
	Run() 
        // ???
	Setup()
        // ???
	Teardown()
        // ???
	Tick()
}
```

```Go
type System interface {
        // ???
	Setup()
        // ???
        Process(entityManager EntityManager) (state int)
        // ???
	Teardown()
}
```

**Comment by setanarut** [comment]
I just realized that the comments are in default_engine.go file. wrong place

**Comment by andygeiss** [comment]
fixed with commit [f49b1df7c1cb9253745af864d5d76eb12ffd0e03](https://github.com/andygeiss/ecs/commit/f49b1df7c1cb9253745af864d5d76eb12ffd0e03)
